sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,143 @@
+     * (e.g. first and last name, or an address)
+     *
+     * @param propertyKey
+     * @param value
+     * @return
+     */
+    fun setProfileAttribute(propertyKey: KlaviyoProfileAttributeKey, value: String): Klaviyo =
+        setProfile(Profile().also { it[propertyKey] = value })
 
     /**
-     * Creates an identify request for the Klaviyo APIs and begins processing.
+     * Debounce timer for enqueuing profile API calls
+     */
+    private var timer: Clock.Cancellable? = null
+
+    /**
+     * Queues a request to identify profile properties to the Klaviyo API
+     *
+     * Any new identifying properties (externalId, email, phone) will be saved to the current
+     * internally tracked profile. Otherwise any identifiers missing from [profile] will be
+     * populated from the current internally tracked profile info.
+     *
      * Identify requests track specific properties about a user without triggering an event
      *
-     * @param properties A map of properties that define the user
+     * @param profile A map of properties that define the user
+     * @return
+     */
+    fun setProfile(profile: Profile): Klaviyo = apply {
+        UserInfo.mergeProfile(profile)
+
+        timer?.cancel()
+        timer = Registry.clock.schedule(Registry.config.debounceInterval.toLong()) {
+            Registry.apiClient.enqueueProfile(profile)
+        }
+    }
+
+    /**
+     * Clears all stored profile identifiers (e.g. email or phone)
+     *
+     * This should be called whenever an active user in your app is removed
+     * (e.g. after a logout)
+     *
+     * @return
      */
-    fun identify(properties: KlaviyoCustomerProperties) {
-        val request = IdentifyRequest(KlaviyoConfig.apiKey, properties)
-        processRequest(request)
+    fun resetProfile(): Klaviyo = apply {
+        UserInfo.reset()
+        Registry.apiClient.enqueueProfile(UserInfo.getAsProfile())
     }
 
     /**
-     * Processes the given [KlaviyoRequest] depending on the SDK's configuration.
-     * If the batch queue is enabled then requests will be batched and sent in groups.
-     * Otherwise the request will send instantly.
-     * These requests are sent to the Klaviyo asynchronous APIs
+     * Queues a request to track an [Event] to the Klaviyo API
+     * The event will be associated with the profile specified by the [Profile]
+     * If profile is not set, this will fallback on the current profile identifiers
+     *
+     * @param event Name of the event metric
+     * @param properties Additional properties associated to the event that are not for identifying the profile
+     * @return
      */
-    private fun processRequest(request: KlaviyoRequest) {
-        if (KlaviyoConfig.networkUseAnalyticsBatchQueue) {
-            request.batch()
-        } else {
-            request.process()
-        }
+    fun createEvent(event: KlaviyoEventType, properties: Event = Event()): Klaviyo = apply {
```

Comment:
TODO - Reviewing Noah's PR and I forgot that we decided in the RFC doc to make Event type part of the `Event` data structure